lib/model/unittest/CResourceLimitTest.cc

```diff
@@ -389,61 +389,121 @@ void CResourceLimitTest::testLargeAllocations() {
         CModelFactory::TFeatureVec features;
         features.push_back(model_t::E_IndividualCountByBucketAndPerson);
         factory.features(features);
-        CModelFactory::TDataGathererPtr gatherer(factory.makeDataGatherer(FIRST_TIME));
-
-        CResourceMonitor resourceMonitor(1.0);
-        resourceMonitor.memoryLimit(std::size_t(70));
-        const maths::CMultinomialConjugate conjugate;
-        ::CMockEventRateModel model(
-            factory.modelParams(), gatherer,
-            factory.defaultFeatureModels(features, BUCKET_LENGTH, 0.4, true), conjugate,
-            CAnomalyDetectorModel::TFeatureInfluenceCalculatorCPtrPrVecVec(),
-            resourceMonitor);
-
-        CPPUNIT_ASSERT_EQUAL(model_t::E_EventRateOnline, model.category());
-        CPPUNIT_ASSERT(model.isPopulation() == false);
-        core_t::TTime time = FIRST_TIME;
-
-        CPPUNIT_ASSERT(resourceMonitor.areAllocationsAllowed());
-
-        // Add some people & attributes to the gatherer
-        // Run a sample
-        // Check that the models can create the right number of people/attributes
-        ::addPersonData(0, 400, time, *gatherer, resourceMonitor);
-
-        CPPUNIT_ASSERT_EQUAL(std::size_t(400), gatherer->numberActivePeople());
-
-        LOG_DEBUG(<< "Testing for 1st time");
-        model.test(time);
-        CPPUNIT_ASSERT_EQUAL(std::size_t(400), gatherer->numberActivePeople());
-        CPPUNIT_ASSERT_EQUAL(std::size_t(400), model.getNewPeople());
-        CPPUNIT_ASSERT_EQUAL(std::size_t(0), model.getNewAttributes());
-        time += BUCKET_LENGTH;
-
-        ::addPersonData(400, 1000, time, *gatherer, resourceMonitor);
-        model.test(time);
-
-        // This should add enough people to go over the memory limit
-        ::addPersonData(1000, 3000, time, *gatherer, resourceMonitor);
-
-        LOG_DEBUG(<< "Testing for 2nd time");
-        model.test(time);
-        LOG_DEBUG(<< "# new people = " << model.getNewPeople());
-        CPPUNIT_ASSERT(model.getNewPeople() > 2700 && model.getNewPeople() < 2900);
-        CPPUNIT_ASSERT_EQUAL(std::size_t(0), model.getNewAttributes());
-        CPPUNIT_ASSERT_EQUAL(model.getNewPeople(), gatherer->numberActivePeople());
-
-        // Adding a small number of new people should be fine though,
-        // as they're allowed in
-        time += BUCKET_LENGTH;
-        std::size_t oldNumberPeople{model.getNewPeople()};
-        ::addPersonData(3000, 3010, time, *gatherer, resourceMonitor);
-
-        LOG_DEBUG(<< "Testing for 3rd time");
-        model.test(time);
-        CPPUNIT_ASSERT_EQUAL(oldNumberPeople + 10, model.getNewPeople());
-        CPPUNIT_ASSERT_EQUAL(std::size_t(0), model.getNewAttributes());
-        CPPUNIT_ASSERT_EQUAL(model.getNewPeople(), gatherer->numberActivePeople());
+
+        {
+            CModelFactory::TDataGathererPtr gatherer(factory.makeDataGatherer(FIRST_TIME));
```

Comment:
How about splitting this test into `testLargeAllocations[Foreground|Background]Persist`? Also, can we share the code better? Seems like we just need to supply different assertion values.